plugins/module_utils/network/sonic/config/poe/poe.py

```diff
@@ -0,0 +1,675 @@
+                                deleted_list.append(filtered_delete)
+            if len(deleted_list) > 0:
+                commands["cards"] = deleted_list
+
+        if want.get("interfaces") is not None and have.get("interfaces") is not None:
+            to_delete_list = have["interfaces"]
+            if len(want["interfaces"]) > 0:
+                to_delete_list = want["interfaces"]
+
+            deleted_list = []
+
+            for interface_d in to_delete_list:
+                for interface_h in have["interfaces"]:
+                    if interface_d["name"] == interface_h["name"]:
+                        if interface_d.keys() == {"name"}:
+                            deleted_list.append(interface_h)
+                        else:
+                            filtered_delete = deepcopy(interface_d)
+                            for setting in interface_d:
+                                if setting not in interface_h or interface_d[setting] != interface_h[setting]:
+                                    del filtered_delete[setting]
+                            if len(filtered_delete) > 1:
+                                deleted_list.append(filtered_delete)
+            if len(deleted_list) > 0:
+                commands["interfaces"] = deleted_list
+
+        if "global" in want and "global" in have:
+            deleted_global_settings = {}
+            if "auto_reset" in want["global"] and "auto_reset" in have["global"] and want["global"]["auto_reset"] == have["global"]["auto_reset"]:
+                deleted_global_settings.update({"auto_reset": have["global"]["auto_reset"]})
+            if "power_mgmt_model" in want["global"] and "power_mgmt_model" in have["global"] and \
+                    want["global"]["power_mgmt_model"] == have["global"]["power_mgmt_model"]:
+                deleted_global_settings.update({"power_mgmt_model": have["global"]["power_mgmt_model"]})
+            if "usage_threshold" in want["global"] and "usage_threshold" in have["global"] and \
+                    want["global"]["usage_threshold"] == have["global"]["usage_threshold"]:
+                deleted_global_settings.update({"usage_threshold": have["global"]["usage_threshold"]})
+            if len(deleted_global_settings) > 0:
+                commands["global"] = deleted_global_settings
+
+        requests.extend(self.make_delete_requests(commands, have))
+        if commands and len(requests) > 0:
+            commands = update_states(commands, "deleted")
+        else:
+            commands = []
+        return commands, requests
+
+    def validate_normalize_config(self, config):
+        '''validates passed in config has values for power_limit and usage_threshold are in range as well as validate against arg spec.
+        passes back config with interface names normalized'''
+        # first removing null values so validate doesn't break
+        config = remove_none(config)
```

Comment:
I added remove_none because remove_empties removes both null values and empty lists/dicts. I wanted a case to allow the playbook writer to clear out the whole list of interfaces or list of cards. Ended up with having the playbook writer pass in an empty list/dict. but that means remove empties would not work since it removes both none and empty and this implementation would need to keep empties inside

Reply:
Thank you for explaining.

Resolved.